pyxem/generators/indexation_generator.py

```diff
@@ -113,28 +140,44 @@ def correlate(
         chosen_function = select_method_from_method_dict(
             method, method_dict, print_help
         )
+        if method in ["fast_correlation", "zero_mean_normalized_correlation"]:
+            # adds a normalisation to library
+            for phase in library.keys():
+                norm_array = np.ones(
+                    library[phase]["intensities"].shape[0]
+                )  # will store the norms
+
+                for i, intensity_array in enumerate(library[phase]["intensities"]):
+                    norm_array[i] = np.linalg.norm(intensity_array)
+                library[phase][
+                    "pattern_norms"
+                ] = norm_array  # puts this normalisation into the library
+
+            matches = signal.map(
+                correlate_library,
+                library=library,
+                n_largest=n_largest,
+                method=method,
+                mask=mask,
+                inplace=False,
+                **kwargs,
+            )
 
-        # adds a normalisation to library
-        for phase in library.keys():
-            norm_array = np.ones(
-                library[phase]["intensities"].shape[0]
-            )  # will store the norms
-
-            for i, intensity_array in enumerate(library[phase]["intensities"]):
-                norm_array[i] = np.linalg.norm(intensity_array)
-            library[phase][
-                "pattern_norms"
-            ] = norm_array  # puts this normalisation into the library
-
-        matches = signal.map(
-            correlate_library,
-            library=library,
-            n_largest=n_largest,
-            method=method,
-            mask=mask,
-            inplace=False,
-            **kwargs,
-        )
+        elif method in ["full_frame_correlation"] :
+            shape = signal.data.shape[1:]
+            size = 2 * np.array(shape) - 1
+            fsize = [next_fast_len(a) for a in (size)]
```

Comment:
If your data is real-valued/complex-valued, you can make use of the second argument.
```python
fsize = [optimal_fft_size(a, not signal.data.dtype.kind == 'c') for a in size]
```